framework/source/class/qx/ui/basic/Image.js

```diff
@@ -694,20 +741,65 @@ qx.Class.define("qx.ui.basic.Image",
       }
 
       // Optimize case for enabled changes when no disabled image was found
-      if (el.getSource() === source) {
+      if (!isFont && el.getSource() === source) {
         return;
       }
 
       // Apply source
       this.__setSource(el, source);
 
-      // Compare with old sizes and relayout if necessary
-      this.__updateContentHint(
-        ResourceManager.getImageWidth(source),
-        ResourceManager.getImageHeight(source)
-      );
+      // Special case for non resource manager handled font icons
+      if (isFont) {
+        var size;
+
+        // Adjust size if scaling is applied
+        if (this.getScale()) {
+          var width = this.getWidth() || this.getHeight() || 40;
+          var height = this.getHeight() || this.getWidth() || 40;
+          size = width > height ? height : width;
+        }
+        else {
+          var font = qx.theme.manager.Font.getInstance().resolve(source.match(/@([^/]+)/)[1]);
+          size = font.getSize();
+        }
+
+        // Default to something definitively numeric if nothing set
+        if (!size) {
+          size = 0;
+        }
+
+        this.__updateContentHint(size, size);
+        el.setStyle("font-size", size + "px");
```

Comment:
*Nearly* redundant. There's some difference between the width/height handlings. Needs.